pkg/nats/stream/producer_client.go

```diff
@@ -30,27 +28,27 @@ type ProducerClient struct {
 	config                             StreamProducerClientConfig
 }
 
-func NewProducerClient(params ProducerClientParams) (*ProducerClient, error) {
+func NewProducerClient(ctx context.Context, params ProducerClientParams) (*ProducerClient, error) {
 	nc := &ProducerClient{
 		Conn:                               params.Conn,
-		ID:                                 params.ID,
 		activeStreamInfo:                   make(map[string]map[string]StreamInfo),
 		activeConnHeartBeatRequestSubjects: make(map[string]string),
 		config:                             params.Config,
 	}
 
-	go nc.heartBeat(context.Background())
+	go nc.heartBeat(ctx)
 
 	return nc, nil
 }
 
-func (pc *ProducerClient) AddConnDetails(ctx context.Context, connDetails *ConnectionDetails) error {
+func (pc *ProducerClient) AddConnDetails(requestSub string, connDetails *ConnectionDetails) error {
 	pc.mu.Lock()
```

Comment:
nit: as we are no longer using nats connection names, it might be better to rename this type and field names to no longer reflect connections. ConnID seems closer to ConsumerID. Also feel free to put these fields directly under `Request` if you can't figure out a better name for `ConnectionDetails`

Reply:
Done